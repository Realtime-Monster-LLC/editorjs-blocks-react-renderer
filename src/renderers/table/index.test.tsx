import React from 'react';
import { create } from 'react-test-renderer';
import Table, { TableBlockData } from '.';

describe('<Table />', () => {
  describe('when receives a table block', () => {
    const data: TableBlockData = {
      header: ['Name', 'Qtd', 'Price'],
      content: [
        ['Kine', '1 pcs', '100$'],
        ['Pigs', '3 pcs', '200$'],
        ['Chickens', '12 pcs', '150$'],
      ],
      footer: ['Total', '16 pcs', '$450'],
      caption: 'The stock on our store.',
    };

    it('renders a <table> tag', () => {
      expect(create(<Table data={data} />).toJSON()).toMatchSnapshot();
    });
  });

  describe('when receives a table block without footer', () => {
    const data: TableBlockData = {
      header: ['Name', 'Qtd', 'Price'],
      content: [
        ['Kine', '1 pcs', '100$'],
        ['Pigs', '3 pcs', '200$'],
        ['Chickens', '12 pcs', '150$'],
      ],
      caption: 'The stock on our store.',
    };

    it('renders a <table> tag without <tfoot>', () => {
      expect(create(<Table data={data} />).toJSON()).toMatchSnapshot();
    });
  });

  describe('when receives a simple table', () => {
    const data: TableBlockData = {
      content: [
        ['Kine', '1 pcs', '100$'],
        ['Pigs', '3 pcs', '200$'],
        ['Chickens', '12 pcs', '150$'],
      ],
    };

    it('renders a <table> tag without <thead>, <tfoot> or <caption>', () => {
      expect(create(<Table data={data} />).toJSON()).toMatchSnapshot();
    });
  });

  describe('when receives a table 2.20 format', () => {
    describe.each([
      [true, 'renders a <table> tag without <tfoot> or <caption>'],
      [false, 'renders a <table> tag without <thead>, <tfoot> or <caption>'],
    ])('withHeadings = %p', (withHeadings, display) => {
      const data: TableBlockData = {
        withHeadings,
        content: [
          ['Kine', 'Pigs', 'Chicken'],
          ['1 pcs', '3 pcs', '12 pcs'],
          ['100$', '200$', '150$'],
        ],
      };

      it(display, () => {
        expect(create(<Table data={data} />).toJSON()).toMatchSnapshot();
      });
    });
  });
});
