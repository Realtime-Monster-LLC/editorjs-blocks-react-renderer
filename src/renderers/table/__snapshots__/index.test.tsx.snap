// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table /> when receives a simple table renders a <table> tag without <thead>, <tfoot> or <caption> 1`] = `
<table>
  
  <tbody>
    <tr>
      <th
        scope="row"
      >
        Kine
      </th>
      <td>
        1 pcs
      </td>
      <td>
        100$
      </td>
    </tr>
    <tr>
      <th
        scope="row"
      >
        Pigs
      </th>
      <td>
        3 pcs
      </td>
      <td>
        200$
      </td>
    </tr>
    <tr>
      <th
        scope="row"
      >
        Chickens
      </th>
      <td>
        12 pcs
      </td>
      <td>
        150$
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`<Table /> when receives a table block renders a <table> tag 1`] = `
<table>
  <caption>
    The stock on our store.
  </caption>
  <thead>
    <tr>
      <th
        scope="col"
      >
        Name
      </th>
      <th
        scope="col"
      >
        Qtd
      </th>
      <th
        scope="col"
      >
        Price
      </th>
    </tr>
  </thead>
  
  <tbody>
    <tr>
      <th
        scope="row"
      >
        Kine
      </th>
      <td>
        1 pcs
      </td>
      <td>
        100$
      </td>
    </tr>
    <tr>
      <th
        scope="row"
      >
        Pigs
      </th>
      <td>
        3 pcs
      </td>
      <td>
        200$
      </td>
    </tr>
    <tr>
      <th
        scope="row"
      >
        Chickens
      </th>
      <td>
        12 pcs
      </td>
      <td>
        150$
      </td>
    </tr>
  </tbody>
  <tfoot>
    <tr>
      <th
        scope="row"
      >
        Total
      </th>
      <td>
        16 pcs
      </td>
      <td>
        $450
      </td>
    </tr>
  </tfoot>
</table>
`;

exports[`<Table /> when receives a table block without footer renders a <table> tag without <tfoot> 1`] = `
<table>
  <caption>
    The stock on our store.
  </caption>
  <thead>
    <tr>
      <th
        scope="col"
      >
        Name
      </th>
      <th
        scope="col"
      >
        Qtd
      </th>
      <th
        scope="col"
      >
        Price
      </th>
    </tr>
  </thead>
  
  <tbody>
    <tr>
      <th
        scope="row"
      >
        Kine
      </th>
      <td>
        1 pcs
      </td>
      <td>
        100$
      </td>
    </tr>
    <tr>
      <th
        scope="row"
      >
        Pigs
      </th>
      <td>
        3 pcs
      </td>
      <td>
        200$
      </td>
    </tr>
    <tr>
      <th
        scope="row"
      >
        Chickens
      </th>
      <td>
        12 pcs
      </td>
      <td>
        150$
      </td>
    </tr>
  </tbody>
</table>
`;
